frontend/src/components/UserSettings/UserGMSSwitch.vue

```diff
@@ -0,0 +1,45 @@
+<template>
+  <div class="form-user-gms-switch">
+    <b-form-checkbox
+      test="BFormCheckbox"
+      v-model="gmsAllowed"
+      name="check-button"
+      switch
+      @change="onSubmit"
+    ></b-form-checkbox>
+  </div>
+</template>
+<script>
+import { updateUserInfos } from '@/graphql/mutations'
+
+export default {
+  name: 'UserGMSSwitch',
+  data() {
+    return {
+      gmsAllowed: this.$store.state.gmsAllowed,
+    }
+  },
+  methods: {
+    async onSubmit() {
+      this.$apollo
+        .mutate({
+          mutation: updateUserInfos,
+          variables: {
+            gmsAllowed: this.gmsAllowed,
+          },
+        })
+        .then(() => {
+          this.$store.commit('gmsAllowed', this.gmsAllowed)
+          this.$emit('gmsAllowed', this.gmsAllowed)
+          this.toastSuccess(
+            this.gmsAllowed ? this.$t('settings.GMS.enabled') : this.$t('settings.GMS.disabled'),
+          )
+        })
+        .catch((error) => {
+          this.gmsAllowed = this.$store.state.gmsAllowed
+          this.toastError(error.message)
+        })
+    },
```

Comment:
Is it correct to say that for the `gmsAllowed` switch the invocation of `updateUserInfos` will only be triggered `on submit` of the page form for example with a _save-_ or _ok-button_ to close the dialog?

Reply:
Nein, schaue weiter oben, onSubmit wird bei jedem @change aufgerufen.
Aber du hast recht, der Name ist etwas irreführend, ich ändere ihn auf onChange.

Reply:
ok